src/libraries/Microsoft.PowerFx.Core/Public/Values/TableValue.cs

```diff
@@ -38,6 +38,16 @@ public abstract class TableValue : ValidFormulaValue
 
         public new TableType Type => (TableType)base.Type;
 
+        /// <summary>
+        /// Retrieves record from the table by a primary key.
+        /// </summary>
+        /// <param name="key">Primary key that can be retrieved via <see cref="RecordValue.TryGetPrimaryKey(out string)"/>.</param>
+        /// <param name="cancellationToken"></param>
+        public virtual async Task<RecordValue> GetRecordAsync(string key, CancellationToken cancellationToken)
```

Comment:
nit to consider:
- Who generates the error message? There's probably a common message we can generate ("record X can't be cast to type Y").  
-  if it is a common message, then this just needs to signal failure (bool TryGetRecord ), and we can fill in the message. 
- else we could return DValue<REcordValue> instead, which would let the caller pass an error message.  Although if we use common one, that's not necessary, 
- Does this need to be async? Can we tell statically whether it will work?


-

Reply:
-Added a common error msg
-Now returns DValue<RecordValue> instead of RecordValue
-it sync operation now.